### PR TITLE
New documentation page for installation and setup of Che on OpenShift

### PR DESCRIPTION
**Pull request by l0rd**
New documentation page with the instructions to deploy Che on different OpenShift flavours:

- OpenShift Container Platform
- minishift
- OpenShift.io

### CODE REVIEW

File: src/main/_docs/setup/setup-openshift.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+DEPLOY_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/deploy_che.sh
+WAIT_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/wait_until_che_is_available.sh
+STACKS_SCRIPT_URL=https://raw.githubusercontent.com/eclipse/che/master/dockerfiles/cli/scripts/openshift/replace_stacks.sh
+curl -fsSL ${DEPLOY_SCRIPT_URL} -o get-che.sh
+curl -fsSL ${WAIT_SCRIPT_URL} -o wait-che.sh
+curl -fsSL ${STACKS_SCRIPT_URL} -o stacks-che.sh
+bash get-che.sh && wait-che.sh && stacks-che.sh
+```
+
+# Deploy Che on openshift.io
```
Comment (bmicklea) [inline src/main/_docs/setup/setup-openshift.md]: Is this something we want users doing?

File: src/main/_docs/setup/setup-openshift.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+You can set different deployment options using environment variables:
+
+* `OPENSHIFT_FLAVOR`: possible values are `ocp`, `minishift` and `osio` (default is `minishift`)
+* `OPENSHIFT_ENDPOINT`: url of the OpenShift API (default is unset for ocp, `https://$(minishift ip):8443/` for minishift, `https://api.starter-us-east-2.openshift.com` for osio)
+* `OPENSHIFT_TOKEN` (default is unset)
+* `CHE_OPENSHIFT_PROJECT`: the OpenShift namespace where Che will be deployed (default is `eclipse-che` for ocp and minishift and `${OPENSHIFT_ID}-che` for osio)
+* `CHE_IMAGE_REPO`: `che-server` Docker image repository that will be used for deployment (default is `docker.io/eclipse/che-server`)
+* `CHE_IMAGE_TAG`: `che-server` Docker image tag that will be used for deployment (default is `nightly-centos`)
+* `CHE_LOG_LEVEL`: Log level of che-server (default is `DEBUG`)
```
Comment (bmicklea) [inline src/main/_docs/setup/setup-openshift.md]: Is there a recommended level or normal usage?  Info?  Warn?

File: src/main/_docs/setup/setup-openshift.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+* `CHE_DEBUGGING_ENABLED`: If set to `true` the script will create the OpenShift service to debug che-server (default is `true`)
+* `CHE_KEYCLOAK_DISABLED`: If this is set to true Keycloack authentication will be disabled (default is `true` for ocp and minishift, `false` for osio)
```
Comment (bmicklea) [inline src/main/_docs/setup/setup-openshift.md]: Not necessary for this release, but shouldn't we be consistent with boolean variables all being `..._ENABLED` versus switching between `ENABLED` and `DISABLED`?

File: src/main/_docs/setup/setup-openshift.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+* `OPENSHIFT_USERNAME`: username to login on the OpenShift cluster. Ignored if `OPENSHIFT_TOKEN` is set (default is `developer`)
+* `OPENSHIFT_PASSWORD`: password to login on the OpenShift cluster. Ignored if `OPENSHIFT_TOKEN` is set (default is `developer`)
+
+# Custom Che Workspace Runtimes in OpenShift
+
+Defining a custom Runtime Stack using [a recipe]({{base}}{{site.links["devops-runtime-recipes"]}}), as a Dockerfile or a Docker Compose file, is currently not supported on OpenShift.
+
+However it is still possible to define new Runtime Stacks using already built Docker images that have been pushed to a Docker registry. Instructions to create a custom Runtime Stack can be found [here]({{base}}{{site.links["devops-runtime-stacks"]}}).
+
+When using a custom Docker image that image must meet [the normal criteria]({{base}}{{site.links["devops-runtime-recipes"]}}#che-runtime-required-dependencies) required to run as a Che runtime. In addition it should follow the [guidelines to creat Docker images to run on OpenShift](https://docs.openshift.org/latest/creating_images/guidelines.html#openshift-origin-specific-guidelines).
```
Comment (bmicklea) [inline src/main/_docs/setup/setup-openshift.md]: Typo: "creat" instead of "create"

File: src/main/_docs/setup/setup-openshift.md (hunk 1 above)
Comment (l0rd) [reply src/main/_docs/setup/setup-openshift.md]: That is how we (che-osio team) currently test before pushing to prod. But no we don't want osio user to do that. We want Che developers to be able to do that when testing a fix in OSIO. If you think that's confusing I can keep this part of the doc in github README https://github.com/redhat-developer/rh-che/blob/master/README.md#deploy-to-openshiftio

File: src/main/_docs/setup/setup-openshift.md (hunk 3 above)
Comment (l0rd) [reply src/main/_docs/setup/setup-openshift.md]: I agree. We already had an issue for that 😄 https://github.com/redhat-developer/rh-che/issues/224

File: src/main/_docs/setup/setup-openshift.md (hunk 4 above)
Comment (l0rd) [reply src/main/_docs/setup/setup-openshift.md]: Thanks, fixed.

File: src/main/_docs/setup/setup-openshift.md (hunk 2 above)
Comment (l0rd) [reply src/main/_docs/setup/setup-openshift.md]: I've changed that to:
```
Logging level of output for Che server. Can be `debug` or `info` (default is `DEBUG`)
```
That's the same sentence used in the Docker installation page 😄

File: src/main/_docs/setup/setup-openshift.md (hunk 1 above)
Comment (bmicklea) [reply src/main/_docs/setup/setup-openshift.md]: Maybe just clarify that for developers testing changes to Che they should do this.

Comment (l0rd) [reply src/main/_docs/setup/setup-openshift.md]: Ok I've added a sentence to make it explicit.

**Pull request merged**